corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -251,12 +261,31 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
         },
 
         getAnswers: function () {
-            var $fields = $(".query-field"),
+            var $inputGroups = $(".query-input-group"),
                 answers = {},
                 model = this.parentModel;
-            $fields.each(function (index) {
-                if (this.value !== '') {
-                    answers[model[index].get('id')] = encodeValue(model[index], $(this).val());
+            $inputGroups.each(function (index) {
+                var queryValue = $(this).find('.query-field').val(),
+                    inputType = model[index].get('input'),
+                    searchForBlank;
+
+                if (inputType === 'select1') {
```

Comment:
I read this section (here to the end of the function) a couple of times before getting it. It seems like multiselects are handled significantly differently from the other field types - it's a bit confusing that `searchForBlank` isn't used for multiselect, and the `isArray` check seems to be a second way of checking `inputType === 'select'`.

I wonder if it would be clearer/simpler if you grouped all of the logic by input type:
```
if (inputType === 'select') {
    ...
    return;
}

var searchForBlank;
if (inputType === 'select1') {
   ...
```